woocommerce/woocommerce-gutenberg-products-block
# Product Gallery: Add keyboard access

## What

Fixes https://github.com/woocommerce/woocommerce-blocks/issues/11284

Adds keyboard access (ESC, LEFT ARROW, RIGHT ARROW) to the Product Gallery overylay/dialog/lightbox.

## Why

By adding keyboard access to the Product Gallery overlay/dialog/lightbox, it makes it more accessible and better UX.

## Testing Instructions



_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. Go to Appearance->Editor->Templates->Single Product (Transform if you're still using classic)
2. Add the Product Gallery block.
3. Go to a product that has gallery images in the frontend.
4. Click on the main image to open up the overlay.
5. Try using your left/right arrow keys to navigate through the different images.
6. Note that you are not able to spam the keys as it has a delay in place to prevent this.
7. Now try pressing the ESC key.
8. Ensure that the overlay closes.


## WooCommerce Visibility


Required:


## Checklist

Required:

Conditional:

## Changelog


> Product Gallery: Add keyboard access
